xbmc/addons/AddonStatusHandler.cpp

```diff
@@ -100,10 +102,10 @@ void CAddonStatusHandler::Process()
     pDialog->SetLine(1, CVariant{24074});
     pDialog->Open();
 
-    CAddonMgr::GetInstance().GetCallbackForType(m_addon->Type())->RequestRestart(m_addon, true);
+    CServiceBroker::GetBinaryAddonManager().OnRequestRestartEvent(m_addon->ID());
   }
   /* Some required settings are missing/invalid */
-  else if ((m_status == ADDON_STATUS_NEED_SETTINGS) || (m_status == ADDON_STATUS_NEED_SAVEDSETTINGS))
```

Comment:
Can you please explain, what NEED_SAVEDSETTINGS formerly was good for and why it is no longer needed?